_source/_posts/2021-12-20-net-azure-cognitive-services.md

```diff
@@ -0,0 +1,829 @@
+---
+layout: blog_post
+title: "Using Azure Cognitive Services in a .NET App"
+author: alisa-duncan
+by: advocate
+communities: [.net]
+description: "This tutorial shows you how to call Azure Cognitive Services to verify faces and store profiles pics in Azure Storage in a .NET MVC app."
+tags: [dotnet, dotnetcore, csharp, aspnet, aspnetcore, dotnet5, azure]
+tweets:
+- "Snap a selfie! 📸  Analyze and detect faces using #azure cognitive services in this #dotnet app!" 
+- "Are you who you say you are? Analyze and detect faces using #azure cognitive services in this #dotnet app! 💻"
+- ""
+image: blog/net-azure-cognitive-services/azure-cognitive-services-social.jpg
+type: conversion
+---
+
+[Azure Cognitive Services](https://azure.microsoft.com/en-us/services/cognitive-services/) is a collection of cloud-based AI products from Microsoft Azure to add cognitive intelligence into your applications quickly. With Azure Cognitive Services, you can add AI capabilities using pre-trained models, so you don't need machine learning or data science experience. Azure Cognitive Services has vision, speech, language, and decision-making services.
+
+In this article, you will learn how to use the Vision service's [Face](https://docs.microsoft.com/en-us/azure/cognitive-services/face/) library to perform facial analysis in a .NET MVC application and store user profile pictures in [Azure Blob Container Storage](https://azure.microsoft.com/en-us/services/storage/blobs/). You'll also authenticate with Okta and store user data as custom profile attributes.
```

Comment:
```suggestion
In this article, you will learn how to use the [Face service](https://docs.microsoft.com/en-us/azure/cognitive-services/face/) library to perform facial analysis in a .NET MVC application and store user profile pictures in [Azure Blob Container Storage](https://azure.microsoft.com/en-us/services/storage/blobs/). You'll also authenticate with Okta and store user data as custom profile attributes.
```
Not seeing any reference to "Vision" per se.

Reply:
@havi I was drawing from Azure Cognitive Services site, where they define service categories (Speech, Language, Vision, Decision) to match with the last sentence of the paragraph above.

But maybe it doesn't have to be that specific and "Face AP"I is enough? (I don't know how I feel about "Face library"). Here's a screenshot of Azure's site with their offerings categorized. Thoughts?

![image](https://user-images.githubusercontent.com/7586237/149039861-3c2fff5e-c62d-4a63-9eaf-df9788c04800.png)

Reply:
"In this article, you will learn how to use the Vision [Face API](https://docs.microsoft.com/en-us/azure/cognitive-services/face/)"

Is my two cents.

Reply:
Thanks both! I think Heather's two cents nails it  - I didn't spot the word "Vision" anywhere when I landed on the "cognitive-services/face/" page so I thought that naming might have changed.